src/relay/op/nn/nn.cc

```diff
@@ -171,6 +171,66 @@ RELAY_REGISTER_OP("nn.leaky_relu")
 .add_type_rel("Identity", IdentityRel);
 
 
+TVM_REGISTER_NODE_TYPE(PReluAttrs);
+
+bool PReluRel(const Array<Type>& types,
+              int num_inputs,
+              const Attrs& attrs,
+              const TypeReporter& reporter) {
+  CHECK_EQ(types.size(), 3);
+  const auto* data = types[0].as<TensorTypeNode>();
+  if (data == nullptr) return false;
+
+  const PReluAttrs* param = attrs.as<PReluAttrs>();
+  CHECK(param != nullptr);
+
+  size_t ndim = data->shape.size();
+  CHECK(ndim == 4)
+    << "Input data should be 4D, but got " << ndim;
+
+  CHECK(reporter->Assert(param->axis < make_const(Int(64), ndim)))
```

Comment:
We should avoid  the abusive use of symbolic Assert, specifically, since param->axis is int, which means we can directly check int relations. Assert is only needed if one of the input expression is symbolic

Reply:
This seems to be a common mistake pattern in recent PRs, can you add more comments to Assert to actually document the possible misuse cases and the intention of the API?